### PR TITLE
Move the remove icon button to the ellipsis menu as a menu button

### LINKED ISSUE #7216: Move block removal under ellipsis
Where block removal is placed right now it's easy to accidentally hit it when you are interacting with the menu. Removal of a block should be a deliberate process and this allows that.

removeblock

### PR DESCRIPTION
**Pull request by talldan**
## Description
Fixes #7216, moves the remove icon button to the ellipsis menu as a menu button.

## How has this been tested?
- Added unit tests for 'Remove Block' menu button
- Performed manual tests including removing single and multiple blocks

## Screenshots
screen shot 2018-06-26 at 11 24 08 am

## Types of changes
- Removed old icon button version of remove button
- Altered existing Remove menu button (which still seemed to be present from previous version) to display like other menu buttons
- Added tests for menu button

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows the accessibility standards
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by karmatosed** [comment]
Thanks for working on this @noisysocks. Could you add a seperator using the style used in other drop downs?

artboard

This visually creates a good divide in actions.

**Comment by talldan** [comment]
Well, I thought adding a separator would be nice and easy, but there were a few intricacies:

- In the other menu, a `MenuGroup` component is used to separate the menu items. The styling was applied only to that menu's `MenuGroup`s though. I've changed it now so that it applied to all `MenuGroup`s. I had a search in the codebase, and the only thing that it might affect that I'm unsure how to view is something called `PluginSidebarMoreMenuItem` - I guess I'll need some plugins to view that?
- I had to refactor a bit, since the menu items for 'Convert to Regular Block' and 'Delete Shared Block' were actually rendered in one parent component and because of that it was impossible to get a `MenuGroup` around the menu items we wanted. I've extracted the Delete Shared Block into its own component.
- Using `MenuGroup`s in the block settings menu introduces a bit of extra padding. It could be removed using a modifier. Let me know.

Here's a couple of screenshots showing how it looks now:
screen shot 2018-06-27 at 5 26 49 pm
screen shot 2018-06-27 at 5 27 13 pm

**Comment by talldan** [comment]
Another very little thing I thought. It feels like when the menu is open, the ellipsis icon button should really not become invisible. In my screenshots above, the popover menu feels like its calling out from nowhere.

**Comment by karmatosed** [comment]
This looks great and thanks for the hardwork wrangling this @talldan. I agree let's have the ellipsis show.

**Comment by noisysocks** [comment]
> I've changed it now so that it applied to all MenuGroups. I had a search in the codebase, and the only thing that it might affect that I'm unsure how to view is something called PluginSidebarMoreMenuItem - I guess I'll need some plugins to view that?

You can test the plugins menu by pasting [this snippet](https://gist.github.com/gziolo/e2954aa83aa1f823b2b05ca1660c2223#file-my-plugin-es5-js) (the ES5 version) into your DevTools console. This will add a button into the More menu. See https://github.com/WordPress/gutenberg/pull/6031 for some context on that feature.

File: editor/components/block-settings-menu/shared-block-delete-button.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/element';
+import { IconButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { isSharedBlock } from '@wordpress/blocks';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+export function SharedBlockDeleteButton( { sharedBlock, onDelete, itemsRole } ) {
```
Comment (noisysocks) [inline editor/components/block-settings-menu/shared-block-delete-button.js]: I like that these components only have one responsibility now 🙂🌈

**Review by noisysocks — CHANGES_REQUESTED** [review]
Great work on this! It's looking really really close.

I'm not sure about re-using `MenuGroup` in this way. I support the general idea but there's a few implications:

1. We now have two `<div role="menu">`s nested within a `<div role="menu">` which feels a little weird to me.

screen shot 2018-06-28 at 10 54 08

2. The child `<div role="menu">`s have an `aria-labelledby` attribute which refers to an identifier that doesn't exist. Admittedly, this is probably a bug in `MenuGroup`.

3. Moving the `border-bottom` rule to `MenuGroup` has caused some regressions with `PluginSidebarMoreMenuItem`, like you suspected it would:

screen shot 2018-06-28 at 10 56 16

4. The extra padding—like you said 🙂

---

I notice that we have a `.editor-block-settings-menu__separator` class defined already, so an alternative to `MenuGroup` might be to simply use a div with that class to add the separator line. You can see an example of us doing exactly that in `editor/components/block-settings-menu/block-transformations.js`.

**Review by noisysocks — APPROVED** [review]

**Comment by hedgefield** [comment]
LOVE THIS.

**Pull request merged**